mlir/lib/Conversion/VectorToArmSME/VectorToArmSME.cpp

```diff
@@ -427,12 +427,88 @@ struct TransposeOpToArmSMELowering
   }
 };
 
+/// Conversion pattern for vector.outerproduct.
+///
+/// If the vector.outerproduct is masked (and the mask from a
+/// vector.create_mask), then the mask is decomposed into two 1-D masks for the
+/// operands.
```

Comment:
And what if it isn't masked?

Reply:
s/vector.outerproduct/arm_sme.outerproduct/g :P